### PR TITLE
Bump bump ic-js

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Bump ic-js which was updated with last did files of the CMC, ledger and index canisters.

# Changes

- bump lib
- update anonymize function to leverage new `approve` type.

# Notes

No particular test added for `approve` and `spender` fields because those are related to ICRC-2 which NNS dapp does not implement yet in any features.

### CODE REVIEW

File: frontend/src/lib/utils/anonymize.utils.ts [hunk 1]
```diff
@@ -321,6 +347,7 @@ const anonymizeSnsTransaction = async (
     burn: await anonymizeTransfer(tx.burn as TransferOpt),
     mint: await anonymizeTransfer(tx.mint as TransferOpt),
     transfer: await anonymizeTransfer(tx.transfer as TransferOpt),
+    approve: await anonymizeApprove(tx.approve as ApproveOpt),
```
Comment (dskloetd) [inline frontend/src/lib/utils/anonymize.utils.ts:350]: Does `approve` need its own anonymizing function?
Burn, mint and transfer all go through the same `anonymizeTransfer`.

File: frontend/src/lib/utils/anonymize.utils.ts [hunk 2]
```diff
@@ -311,6 +311,32 @@ const anonymizeTransfer = async (
...
+
+  return {
+    to: await anonymizeSnsAccount(data.to),
+    from: await anonymizeSnsAccount(data.from),
+    spender: await anonymizeSnsAccount(data.spender),
+    fee: data.fee,
+    memo: data.memo,
+    created_at_time: data.created_at_time,
+    amount: await anonymizeAmount(data.amount),
+    expected_allowance: data.expected_allowance,
```
Comment (dskloetd) [inline frontend/src/lib/utils/anonymize.utils.ts:336]: Isn't this like an amount that should be anonymized like an amount?

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/utils/anonymize.utils.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/anonymize.utils.ts:350]: The `approve` field have few more types, three to be precise.

File: frontend/src/lib/utils/anonymize.utils.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/anonymize.utils.ts:336]: Sure good idea.

File: frontend/src/lib/utils/anonymize.utils.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/lib/utils/anonymize.utils.ts:350]: `Burn` also has a `spender` field, for example, but apparently we don't care:
https://github.com/dfinity/ic-js/blob/4c8dd0b22416f86f75b5a957fa6a210b44fd5156/packages/ledger/candid/icrc1_index.d.ts#L23

Fine with me either way, but I was confused by it when I was trying to fix it myself.

Comment (peterpeterparker) [reply frontend/src/lib/utils/anonymize.utils.ts]: I don't mind neither so I removed the new code and used `anonymizeTransfer`.

File: frontend/src/lib/utils/anonymize.utils.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/anonymize.utils.ts]: Code has been removed so...

**Pull request merged**